### ISSUE TITLE
Problems with empty attributes

### ISSUE THREAD

**Issue by richtera** [pveyes/htmr#137]
https://github.com/pveyes/htmr/blob/b691b2d52dc26af1f0fdf7235dabbc01752bcb57/src/mapAttribute.ts#L49

Hi the way this detects truthy attributes does make sense, but since the definition for the RawProperties doesn't allow for boolean values, it gets type casted to a string with the value "true"

https://github.com/pveyes/htmr/blob/b691b2d52dc26af1f0fdf7235dabbc01752bcb57/src/mapAttribute.ts#L3

This is causing problems parsing and rendering some things. For example
`<img alt="" src="Something"/>` will end up with alt="true" and if you substribute a react component on top it will complain that alt="true' ie. `Warning: Received `true` for a non-boolean attribute `alt``
Thanks
Andy

**Comment by pveyes** [comment]
The return value of `mapAttribute` is `Attributes` type, not `RawAttribute`, and it includes `boolean`. 
https://github.com/pveyes/htmr/blob/132769d1473c710c6b016bc6d92ab593bfbebdd2/src/mapAttribute.ts#L9 This is also internal API that's not exposed so I don't know how it's relevant

> This is causing problems parsing and rendering some things. For example
>
> will end up with alt="true" and if you substribute a react component on top it will complain that alt="true' ie. Warning: Received truefor a non-boolean attributealt``

Can you elaborate more on this? I think there's a snippet of code that's get removed by GH because you're not using a code block.

As for `"true"` value, it should be passed as is as a string. You can see in this test PR https://github.com/pveyes/htmr/pull/138 that it keep the string as is without any warning.

If you can send a failing test case that would be great

**Comment by richtera** [comment]
Ok I stand corrected on that, but I think it's still causing a problem.
During debugging I see: 
![image](https://user-images.githubusercontent.com/708186/128330222-01a0dac6-625f-4769-acd2-1b7579fdc5f5.png)
And then I still get the failure. I can work around it then, but it does seems weird that if I don't use a custom renderer on img it correctly translates the alt="" to an empty property.
I'll add a workaround ignoring alt === true and removing it from the props.
Something like this for the "transform" component would work:
```javascript
function Image({ alt, ...props }) {
  const zoomClick = useContext(LightboxContext);
  if (alt !== true) {
    props.alt = alt;
  }
  return <img {...props} onClick={zoomClick} onDoubleClick={zoomClick} />;
}
```
Feels awkward, but I agree it's not "wrong" and does work.

**Comment by pveyes** [comment]
Maybe I misunderstood something, but if your HTML is `<img alt="" />` the react element equivalent would be `<img alt={true} />`, or if you're using custom transform, the `props.alt` would be `true`.

Based on HTML spec, this is expected behavior. If you don't want to get the warning from React, you need to remove all empty string for non-boolean attribute in the HTML, or using custom transform to remove invalid boolean attribute

**Comment by pveyes** [comment]
I think I know the issue here, not all attributes are eligible as boolean attribute, so it should not be treated as such

I will find a way to check whether attribute is a boolean attribute without increasing bundle size too much because hardcoding a list is not preferable

**Comment by pveyes** [comment]
Published as `v1.0.1`

Thanks for raising this issue!
